src/FormElement/TermInput.php

```diff
@@ -268,19 +269,19 @@ public function prepareMultipartUpdate(ServerRequestInterface $request): array
         if ($this->valueHasBeenPasted()) {
             $updates[] = $this->termContainer();
             $updates[] = [
-                HtmlString::create(json_encode(['#' . $this->getName() . '-search-input', []])),
+                HtmlString::create(json_encode(['#' . Attribute::sanitizeId($this->getName()) . '-search-input', []])),
```

Comment:
This might still not address the duplicate ID issue, if the term inputs are added to different fieldsets with the same name in the same form. In this case both the term inputs receives the same ID. That was the reason I had previously used `getValueOfNameAttribute` instead.